src/the-nest/annotations.py

```diff
@@ -0,0 +1,37 @@
+x: int
+y: list[int | float]
+
+class Base:
+    a: int
+    b: str
+
+
+class Derived(Base):
+    pass
+
+
+def pretty_bin(n: int, *, sep: str = " ") -> str:
+    binary = f"{n:b}"
+    while len(binary) % 4 != 0:
+        binary = f"0{binary}"
+    return sep.join(binary[i:i+4] for i in range(0, len(binary), 4))
```

Comment:
```suggestion
    return sep.join("".join(s) for s in zip(*[iter(binary)] * 4))
```
This one is a bit of a stretch, though.
Based it on the "grouper" itertools recipe: https://docs.python.org/3/library/itertools.html#itertools-recipes

Reply:
I don't like this one, feels harder to read